src/tests/e2e/login.post.test.ts

```diff
@@ -0,0 +1,88 @@
+import 'dotenv/config';
+import request from 'supertest';
+import app from '../../app';
+import { mockPassword, mockUsername, mockBasicAuthString } from '../mockObjects/mockUser';
+import { setupE2E } from '../setupTests';
+import { ReasonPhrases, StatusCodes } from 'http-status-codes';
+import { userService } from '../../services/userService';
+import { mockComparePassword, mockGenerateTokens, mockGetUser } from '../mockObjects/mockUserService';
+
+const ENDPOINT = '/login';
+
+setupE2E('login-e2e', () => {
+  const generateTokensSpy = jest.spyOn(userService, 'generateTokens').mockImplementation(mockGenerateTokens);
+  const getUsersSpy = jest.spyOn(userService, 'getUser').mockImplementation(mockGetUser);
+  const comparePasswordSpy = jest.spyOn(userService, 'comparePassword').mockImplementation(mockComparePassword);
+  describe('POST /login responds', () => {
+    test('with set of tokens for a logged in user', async () => {
+      const res = await request(app)
+        .post(ENDPOINT)
+
+        .set('Authorization', mockBasicAuthString(mockUsername, mockPassword));
+      expect(res.statusCode).toBe(StatusCodes.OK);
+      expect(res.body.accessToken).toBeDefined();
+      expect(res.body.refreshToken).toBeDefined();
+    });
+
+    test(`with ${StatusCodes.UNAUTHORIZED} when username is not provided`, async () => {
+      const res = await request(app).post(ENDPOINT).set('Authorization', mockBasicAuthString('', mockPassword));
+      expect(res.statusCode).toBe(StatusCodes.BAD_REQUEST);
+    });
+
```

Comment:
I don't see a test for every case. I.e you check if password is too short, but you don't check if password is too long
<img width="754" alt="image" src="https://user-images.githubusercontent.com/59976377/183412357-7980acb1-4cbc-41c5-83bc-47339b4d4716.png">

By the way I found that you check username length in password validation => should be changed to password.

Reply:
And I would wrap validation tests in another `describe()` in order to organise them, because now they are mixed between status code tests